examples/README.md

```diff
@@ -1,7 +1,23 @@
 # IBM Analog Hardware Acceleration Kit: Examples
 
 We have many different examples to explore the many features of the IBM Analog Hardware
-Acceleration Kit:
+Acceleration Kit.
+
```

Comment:
Thanks for adding test-related information - can it be moved to a more sensible place, though? Usually, running the examples is a different use case than running the tests, which in my eyes is closer to contributing.

One of these places might be [`development_conventions.rst`](https://github.com/IBM/aihwkit/blob/master/docs/source/development_conventions.rst#testing) - there is a `Testing` section that can be expanded (and a mention to `pytest` earlier on). For more visibility, it would also make sense to add a link to that documentation page from the [contribution guidelines](https://github.com/IBM/aihwkit/blob/master/.github/CONTRIBUTING.md): that would keep the initial contribution lean, and still allow users to reach the full development conventions.

Reply:
Good point. I agree we should only show here how to run the examples rather than the unit tests. what do you think of moving this section to the redthedoc documentation? @diego-plan9

Reply:
My suggestion would be:
* `examples/README.md` - keeping it the same as `master`, with information on how to run the examples.
* new information about running tests - add to `development_conventions.rst`, so it does end up in readthedocs [here](https://aihwkit.readthedocs.io/en/latest/development_conventions.html).
* additionally, add a link to [the rendered development conventions](https://aihwkit.readthedocs.io/en/latest/development_conventions.html) to `CONTRIBUTING.md`

The thinking is that running the tests is usually valuable for users interested in contributing or exploring the code from a developer perspective - and github makes it easy to have the `CONTRIBUTING.md` as the natural starting point (it is automatically mentioned when creating a PR for the first time, in the [community standards](https://github.com/IBM/aihwkit/community), etc). From that starting point, having the link to the development conventions would allow for easily discovering not only the test needs, but other pieces of information that are likely to be useful to that user group.

It's just a suggestion, though - having the information in `tests/README.md` is not extremely uncommon either (arguably a bit harder to incorporate into the overall documentation flow, and also blurring slightly the purpose of the `tests/` folder, which is mostly for the test themselves). The important part is that the information ends up in the repository one way or the other :+1: